src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4400,21 +4403,23 @@ IsPartitionColumn(Expr *columnExpression, Query *query)
 /*
  * FindReferencedTableColumn recursively traverses query tree to find actual relation
  * id, and column that columnExpression refers to. If columnExpression is a
- * non-relational or computed/derived expression, the function returns InvalidOid for
- * relationId and NULL for column. The caller should provide parent query list from
+ * non-relational or computed/derived expression, the function returns NULL for
+ * rte and NULL for column. The caller should provide parent query list from
  * top of the tree to this particular Query's parent. This argument is used to look
  * into CTEs that may be present in the query.
+ *
+ * If skipOuterVars is true, then it doesn't check vars coming from outer queries.
```

Comment:
Would be nice to add a comment why we made this optional. I think we can openly write that it should just work, but as the change touches wide range of query classes, we wanted to work on that separately, make this false for all, and remove this flag.